extensions/variance/src/main/java/io/druid/query/aggregation/variance/VarianceBufferAggregator.java

```diff
@@ -0,0 +1,182 @@
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.variance;
+
+import com.google.common.primitives.Doubles;
+import com.google.common.primitives.Longs;
+import io.druid.query.aggregation.BufferAggregator;
+import io.druid.segment.FloatColumnSelector;
+import io.druid.segment.LongColumnSelector;
+import io.druid.segment.ObjectColumnSelector;
+
+import java.nio.ByteBuffer;
+
+/**
+ */
+public abstract class VarianceBufferAggregator implements BufferAggregator
+{
+  private static final int COUNT_OFFSET = 0;
+  private static final int SUM_OFFSET = Longs.BYTES;
+  private static final int NVARIANCE_OFFSET = SUM_OFFSET + Doubles.BYTES;
+
+  protected final String name;
+
+  public VarianceBufferAggregator(String name)
+  {
+    this.name = name;
+  }
+
+  @Override
+  public void init(final ByteBuffer buf, final int position)
+  {
+    buf.putLong(position + COUNT_OFFSET, 0)
+       .putDouble(position + SUM_OFFSET, 0)
+       .putDouble(position + NVARIANCE_OFFSET, 0);
+  }
+
+  @Override
+  public Object get(final ByteBuffer buf, final int position)
+  {
+    VarianceHolder holder = new VarianceHolder();
+    holder.count = buf.getLong(position);
+    holder.sum = buf.getDouble(position + SUM_OFFSET);
+    holder.nvariance = buf.getDouble(position + NVARIANCE_OFFSET);
+    return holder;
+  }
+
+  @Override
+  public float getFloat(ByteBuffer buf, int position)
+  {
+    throw new UnsupportedOperationException("VarianceBufferAggregator does not support getFloat()");
+  }
+
+  @Override
+  public long getLong(ByteBuffer buf, int position)
+  {
+    throw new UnsupportedOperationException("VarianceBufferAggregator does not support getFloat()");
+  }
+
+  @Override
+  public void close()
+  {
+  }
+
+  public static final class FloatInput extends VarianceBufferAggregator
+  {
+    private final FloatColumnSelector selector;
+
+    public FloatInput(String name, FloatColumnSelector selector)
+    {
+      super(name);
+      this.selector = selector;
+    }
+
+  @Override
+  public void aggregate(ByteBuffer buf, int position)
+  {
+    float v = selector.get();
+    long count = buf.getLong(position + COUNT_OFFSET) + 1;
+    double sum = buf.getDouble(position + SUM_OFFSET) + v;
+    buf.putLong(position, count);
+    buf.putDouble(position + SUM_OFFSET, sum);
+    if (count > 1) {
+      double t = count * v - sum;
```

Comment:
And supposedly the original algorithm comes from here: http://doi.org/10.1080/00401706.1971.10488826 which is stuck behind a pay wall